pkg/chunk/storage/factory.go

```diff
@@ -164,13 +165,14 @@ func NewIndexClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chun
 }
 
 // NewObjectClient makes a new ObjectClient of the desired types.
-func NewObjectClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.ObjectClient, error) {
+func NewObjectClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.Client, error) {
+	var store chunk.ObjectClient
+	var err error
 	switch name {
```

Comment:
To me, it looks a bit unclear this code now, because we have some `case` which `return` and other `case` which save `store` and `err`. Moreover, in case of error I believe we should not return an instance of `object.ChunkClient`.

A cleaner solution may be adding a function like:
```
func newChunkClientFromStore(store, err) (*object.ChunkClient) {
  if err != nil {
    return nil, err
  }

 return &object.ChunkClient{
		Store: store,
	}, nil
}
```

This way you can use it like:
`return newChunkClientFromStore(aws.NewS3ObjectClient(cfg.AWSStorageConfig.S3Config))`

Reply:
LGTM